src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -71,10 +72,7 @@ private static async Task<HttpResponseMessage> SendWithNtAuthAsync(HttpRequestMe
                             connectionPool.IncrementConnectionCount();
                             connection.Acquire();
                             isNewConnection = true;
-                        }
-                        else
-                        {
-                            await connection.DrainResponseAsync(response).ConfigureAwait(false);
+                            needDrain = false;
```

Comment:
> Shouldn't this first 401 response from the server be completely read (drained) before it sends out the first reply to the 401 with a client-generated blob?

It will be drained, as long as the server doesn't send Connection: close, in which case we don't need to drain because we are establishing a new connection.

Reply:
>It will be drained, as long as the server doesn't send Connection: close

I must be missing something.  If the server doesn't send Connection:close, then the `needDrain` gets set to false.  And then it enter the while(true) loop.  So, where/how does it get drained?

Reply:
Actually, I'm reading the diff wrong. :)

So, forget what I said. I just stepped thru this in the debugger and it looks ok. :)